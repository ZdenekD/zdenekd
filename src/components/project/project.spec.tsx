import {shallow, ShallowWrapper} from 'enzyme';
import findComponent from '../../__test__/utils/helpers';
import Project from '.';

describe('Project', () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = shallow(<Project />);
    });

    it('match snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('renders without error', () => {
        const component = findComponent(wrapper, 'component-project');

        expect(component.exists()).toBe(true);
    });
});
