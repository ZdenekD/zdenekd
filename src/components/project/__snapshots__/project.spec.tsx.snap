// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project match snapshot 1`] = `
<div class="block"
     data-test="component-project"
>
  <header class="content">
    <h2 class="title"
        data-title="Tomton"
    >
      Tomton
    </h2>
    <p class="description"
       data-title="Společnost vyrabějící designové otopné systémy. Jeden z modelů získal celosvětově uznávané ocenění Red Dot Design Award."
    >
      Společnost vyrabějící designové otopné systémy. Jeden z modelů získal celosvětově uznávané ocenění Red Dot Design Award.
    </p>
    <div class="toolsWrapper">
      <ul class="tools"
          data-test="component-tools"
      >
        <li class="tool">
          <figure class="figure">
            <img src="/logo_react.svg"
                 width="32"
                 height="32"
                 alt="Technologie použitá v projektu: react"
                 class="image"
            >
            <figcaption class="caption">
              react
            </figcaption>
          </figure>
        </li>
        <li class="tool">
          <figure class="figure">
            <img src="/logo_wordpress.svg"
                 width="32"
                 height="32"
                 alt="Technologie použitá v projektu: wordpress"
                 class="image"
            >
            <figcaption class="caption">
              wordpress
            </figcaption>
          </figure>
        </li>
        <li class="tool">
          <figure class="figure">
            <img src="/logo_lottie.svg"
                 width="32"
                 height="32"
                 alt="Technologie použitá v projektu: lottie"
                 class="image"
            >
            <figcaption class="caption">
              lottie
            </figcaption>
          </figure>
        </li>
        <li class="tool">
          <figure class="figure">
            <img src="/logo_animejs.svg"
                 width="32"
                 height="32"
                 alt="Technologie použitá v projektu: animejs"
                 class="image"
            >
            <figcaption class="caption">
              animejs
            </figcaption>
          </figure>
        </li>
        <li class="tool">
          <figure class="figure">
            <img src="/logo_webpack.svg"
                 width="32"
                 height="32"
                 alt="Technologie použitá v projektu: webpack"
                 class="image"
            >
            <figcaption class="caption">
              webpack
            </figcaption>
          </figure>
        </li>
      </ul>
    </div>
    <ul class="controls"
        data-test="component-controls"
    >
      <li class="control">
        <button type="button"
                class="prev disabled"
                aria-label="Předchozí projekt"
                data-test="component-controls-prev"
        >
        </button>
      </li>
      <li class="control">
        <button type="button"
                class="next "
                aria-label="Následující projekt"
                data-test="component-controls-next"
        >
        </button>
      </li>
    </ul>
  </header>
  <section class="browser"
           data-test="component-browser"
  >
    <header class="header">
      <span class="controls">
        <i class="control">
        </i>
        <i class="control">
        </i>
        <i class="control">
        </i>
      </span>
      <button type="button"
              class="prev disabled"
              aria-label="Předchozí projekt"
              tabindex="0"
              data-test="component-controls-prev"
      >
      </button>
      <button type="button"
              class="next "
              aria-label="Následující projekt"
              tabindex="0"
              data-test="component-controls-next"
      >
      </button>
      <div class="addressbar https"
           data-test="component-addressbar"
      >
        <a href="https://www.tomton.cz"
           target="_blank"
           rel="noreferrer noopener"
           data-test="component-addressbar-link"
           aria-label="Otevřít referenci v novém okně"
        >
          https://www.tomton.cz
        </a>
      </div>
      <div class="target">
        <a href="https://www.tomton.cz"
           class="anchor targetLink"
           target="_blank"
           rel="noreferrer noopener"
           data-test="component-target"
           aria-label="Otevřít referenci v novém okně"
        >
          <svg data-filename="src/assets/images/icon_open.svg">
          </svg>
        </a>
      </div>
    </header>
    <picture class="picture"
             data-test="component-picture"
    >
      <source srcset="/image/mobile_tomton.avif"
              type="image/avif"
      >
      <source srcset="/image/mobile_tomton.webp"
              type="image/webp"
      >
      <img src="/image/mobile_tomton.png"
           width="480"
           height="270"
           alt
           loading="lazy"
           decoding="async"
           class="image "
      >
    </picture>
  </section>
</div>
`;
